### PR TITLE
Add small info to specify right inlet on log~ helpfile

### PR DESCRIPTION
**Pull request by abreubacelar**
Just a small commit to test git workflow, it adds the word "right" to specify the right inlet on log~ helpfile

### CODE REVIEW

**Comment by porres** [comment]
what do you want to test?

**Comment by abreubacelar** [comment]
> what do you want to test?

i was trying to understand how to contribute to your documentation branch without having to fork your repository

lucas said something that we would need to fork your repository to contribute, but i didn't need it...

it is possible to fork the main repository and add your repository as a remote just to fetch/pull your branch... but i'm still noob at git too, so i did the test to understand how it works, now i can access both htmldocs from the main repository and your Documentation branch in a single local repository, no need to have two separate directories... maybe the problem would be if i had to compile pd? i have to test this too xD

**Comment by porres** [comment]
you know what, I though it wasn't necessary :) 

hey @lucarda check this out

**Comment by Lucarda** [comment]
@abreubacelar 
> i was trying to understand how to contribute to your documentation branch without having to fork your repository

IIRC you where previously forking @porres repo right? I took the idea from you (or from someone that did that).

> it is possible to fork the main repository and add your repository as a remote just to fetch/pull your branch...

can you share how?

**Comment by abreubacelar** [comment]
> @abreubacelar
> 
> > i was trying to understand how to contribute to your documentation branch without having to fork your repository
> 
> IIRC you where previously forking @porres repo right? I took the idea from you (or from someone that did that).
> 

Yes, i was forking porres repository directly, but as i said i am noob at git (now i decided to learn it, so i am taking first steps), so what i did before was all inside github site, so to edit porres files i had to fork his repository and manually copy paste the files to make the PRs xD

> > it is possible to fork the main repository and add your repository as a remote just to fetch/pull your branch...
> 
> can you share how?

so now what i did was something like that (i will give full report of what i remember, but maybe some steps may be wrong or have a better way of doing, remember that i am learning and experimenting in the process): 

First i deleted the repository i had. And forked the main repository, 

from my repository i did the git clone

to access the htmldocs branch, i then did 

git checkout origin/htmldocs
git switch -c htmldocs origin/htmldocs

this makes a new branch on my local repository, right?

then as a failed step, i also tried to fork porres repository again, but interestingly github prevented me from doing that, it says that i couldn't make the repository because i already had abreubacelar/pure-data

then researching about the remotes, i tried this>

git remote add porres https://github.com/porres/pure-data.git
git fetch porres
git checkout porres/Documentation
git switch -c Documentation

this way when i do "git branch -a" i have all origin/<branches> and porres/<branches>, so i can do the fetch specifically from porres

so i made the changes to this file, committed, and then did

git push origin Documentation

Then from my repository i created the pr on the github site, targeting porres branch

(there was more steps as i named the commit wrongly, so i force pushed as you can see in the history)

as an additional research step, git remote add has also a -t flag to specify an specific branch to track, I then did 

git remote add -t kbdnav HenriAugusto https://github.com/HenriAugusto/pure-data.git
git fetch HenriAugusto
git checkout HenriAugusto/kbdnav
git switch -c kbdnav

this made the "git branch -a" different, in this case i have only  "remotes/HenriAugusto/kbdnav" from HenriAugusto, while i have all "remotes/porres/<branches>" so i think using -t flag would be cleaner... i will check later to remove porres and do this way again

So basically my workflow now would be to fetch/pull from all the remotes, push to origin and then target the PR to the correct repository/branch here... from a single local repository...

**Comment by Lucarda** [comment]
@abreubacelar 
Thanks for the explanation. I'm also learning.

from what you said and if you didn't do real work but just testing can you try (its totally ok that you say no to this :-) :

- delete your fork.

- create (again) a new fork from the main pure-data 

then


`git remote add -t kbdnav HenriAugusto https://github.com/HenriAugusto/pure-data.git`

`git fetch HenriAugusto`

`git checkout HenriAugusto/kbdnav`

i think here you will want/need to do `git pull` [1]

`git switch -c kbdnav` i think you can give it a different name here to avoid confusion 


[1] Is git fetch and pull same?
git fetch is the command that tells your local git to retrieve the latest meta-data info from the original (yet doesn't do any file transferring. It's more like just checking to see if there are any changes available). git pull on the other hand does that AND brings (copy) those changes from the remote repository.

|Git Fetch |Git Pull|
|:--|:--|
|Gives the information of a new change from a remote repository without merging into the current branch|Brings the copy of all the changes from a remote repository and merges them into the current branch|
|Repository data is updated in the .git directory |The local repository is updated directly|
|Review of commits and changes can be done|Updates the changes to the local repository immediately.|
|No possibility of merge conflicts.|Merge conflicts are possible if the remote and the local repositories have done changes at the same place.|

**Comment by Lucarda** [comment]
then of course then you can do the same with `porres:Documentation`

**Comment by abreubacelar** [comment]
i did some tests with a new repo, and what i learned is that to create the branch both fetch and pull works, but pull yells at me xD

it gives an error saying that i have no branch to merge... and it is true, since i am creating the branch so the branch doesn't exist yet

so the correct steps are indeed

git add remote newRemote url/to/newRemoteOriginalName
git fetch newRemote
git checkout newRemote/branch
git switch -c nameOfNewBranchOnMyRepository

checkout does the job of going into a detached branch with all the files of the remote branch, but it only works if i do fetch or pull before. So fetch only gets the info to get ready to do checkout, while pull does fetch and then gives the error. But since pull did fetch the next checkout works.

git switch -c then copies this detached branch into my local repo as a new branch, after that i can do fetch or pull, but to be honest i need real changes in the repos to see pull working xD

as i understood in the tutorials, git pull is the same as git fetch + git merge or git fetch + git rebase, this is new to me yet, i need to understand both

btw, as a related test, i can instead of cloning a repo, i can do the same starting with git init

so instead of cloning>

git init
git add remote etc...
git fetch ...
git checkout ...
git switch -c ...

does the same as the git clone (at least to what i experimented, and understand). But of course this isn't necessary, git clone is less work, so it is faster xD

**Comment by Lucarda** [comment]
@abreubacelar thank for your findings.

this makes `pull` work. 

`git remote add -t Documentation porresdoc https://github.com/porres/pure-data.git`
`git fetch porresdoc`
`git checkout porresdoc/Documentation`
`git switch -c docporres`

`git pull` (this time it does not work so do next)

`git branch --set-upstream-to=porresdoc/Documentation docporres`

`git pull` (this time we are OK)


Needs to be tested in real world situation when one has already started a trackin branch and that branch has gone ahead.

**Comment by porres** [comment]
I've merged this anyway...

You can keep discussing here of course :)

So, is it ok now that I have an html branch of mine?

I see the one upstream still getting PRs and all... I was hoping to concentrate the html changes and merge it to my main 'documentation' branch. IOhannes didn't say anything, let's try again?

**Comment by Lucarda** [comment]
our life will be easier if we keep updating html to that `pure-data:htmldocs`

why you want to merge it?

**Comment by abreubacelar** [comment]
i agree with Lucas, i would prefer to update html in pure-data:htmldocs, it also makes easier for more people to help and also propose new features, to be honest I don't understand why we need to contribute here even for patches... main repo could have some helpfile branch too

**Comment by abreubacelar** [comment]
having said that, you can have your html branch, but also fetching/pulling from others on main repo, making your changes and making a pull request there

why this would be difficult?

**Comment by Lucarda** [comment]
I also add to @abreubacelar what says for html that please try https://github.com/pure-data/pure-data/pull/1822

you can download the repo as a zip of that branch https://github.com/Lucarda/pure-data/tree/hugo an unzip it on some folder

it's meant to do the manual editing easier. and you (@porres) are the main editor that comes to my mind.

at least tell me what you don't like about it.

**Comment by porres** [comment]
>  to be honest I don't understand why we need to contribute here even for patches... 

Because I created it and I've been doing most of the work in it with hundreds of commits for a few years now... it makes sense to have a single PR that does the same thing and sometimes we have conflicting PRs that end up being "merged", or one deleted in favour of the other (we all agree right?). It is also quite common to contribute to other peoples' PRs in their repository... I've done it before. I should ask honestly what is the problem of people contributing to my branch?

Ok, I see maybe Pd could have its documentation branch, but if we're gonna make a case out of it, I would really like me to have managing/curating powers on it. The ones who have super powers and are able to create such branches upstream haven't created such branches because they weren't (and aren't) working on it. The only real and separate branch is iohannes' develop and it's mostly his stuff...

Anyway, this discussion would make more sense if lots of people were actually eager to help and are contributing to my work, which I would really love if it were true by the way. One way or another, if people wanna make a case out of having trouble to contributing to my PRs, I did ask that I'd be granted access powers in such cases. If there's no opposition to that, let's do it and everybody is happy...

**Comment by porres** [comment]
> at least tell me what you don't like about it.

haven't tried it yet and, seriously, until I either have my own html branch or editing powers, I'm not bothering :)

**Comment by porres** [comment]
> our life will be easier if we keep updating html to that pure-data:htmldocs
> 
> why you want to merge it?

I honestly don't have a better argument than saying, hey people, I'm really busting my ass off here on this one and would like it to be easier on me as the main editor/manager of documentation updates. My argument is the same as yours, it would make things easier for me... the difference is I've been doing most of the work. This would also make not only my life easier, but also IOhannes' and Miller's and whoever else manages the main repository who is not as much committed to this as I am. Unless.......... maybe I could, again, have editing powers and everyone is happy...

I have already been working on manual updates for a while now as part of my documentation branch. I have, for instance, rewritten a whole new chapter on how to manage externals. I don't think we need to have two separate efforts for it. I'm as committed to working on the manual as I am to working on pd files and even other resources that we've been discussing in 'pddp'... in short, I'm committed to and have been working on Pd's documentation in general. 

I just thought of having a separate html branch that I could eventually merge to my main documentation branch and I don't know and yet don't understand why such a branch was created independently...

It was not the first time I created a separate branch. I did that when creating the 'pd reference' subpatches for all help files and did merge it to the documentation branch when it was ready.

I just have bigger plans now to work on the manual other than typos and would like to write yet new sections, add an html reference and thing like that...

It's nice more people want to help with html files than help files and this is now becoming a case, but please, I think this will severely disrupt the workflow by involving more bureaucracy and third party people that are not as committed to this just so a couple more people contribute a bit more in an easier way...

It would make sense if IOhannes was actually also working on a rewrite of the manual... but it would be ridiculous forcing him to keep merging my rewrite commits just because, I don't know, 3 or 4 people tops would feel it's better to send typo updates or suggestions to me directly...

**Comment by Lucarda** [comment]
> I just thought of having a separate html branch that I could eventually merge to my main documentation branch and I don't know and yet don't understand why such a branch was created independently...

I had asked for `pure-data:htmldocs` and I got it. But this is now on your side as you only need to merge `pure-data:htmldocs` to your html branch (but don't do it yet).

> at least tell me what you don't like about it.

> haven't tried it yet and, seriously, until I either have my own html branch or editing powers, I'm not bothering :)

Bad. you should bother to check the HUGO thing.

**Comment by porres** [comment]
You know, I'm sorry to feel this way, but if working on a centralized upstream branch that I have no direct access to is what is needed, I could probably just take a break from working on the documentation anyway and let you people have fun... I do have other things to concentrate on, so go ahead and do changes to html upstream or even the help files on a new help files branch... I may get back when I see changes have been made and accepted and not much else is going on.

No one is helping me revise the documentation of ELSE and it is much bigger than vanilla's. It's also time I'd sit and work on a huge revision of my tutorial and turn it into a proper book...

**Comment by abreubacelar** [comment]
well, you are right about doing more work... i just don't like the idea of the work not being discussed... 

the reference patches for example, maybe they could go on iohannes direction of being parsable, so instead of doing the work on the patches, we could have some way to automate the creation of reference patches, it would be more maintainable and even could be translated into other media easily. We could change the script in one archive and get a reference patch, and an html version

I remember he talking something like that about your changes, and you just said that you would complete it and also didn't bother about alternatives... 

i don't have right on saying anything, i hoped i could have helped this year but my life gone all difficult and different... i am slowly coming back to a more normal setup and i would like i could help more. in this you are absolutely right, you do more work, thank you for this btw...

so one question, help files could stay here as you say, we have this problem about merging pd files, html also have this problem?

**Comment by abreubacelar** [comment]
about html, i have some ideas about tooltips, for example some section talking about number boxes, if you pass your mouse over the word, it shows an image of a number box (or maybe a gif?) so then we would need to change css, and even think how to deal when the user is on mobile

but also i have another idea to use html area and map tags (https://www.w3schools.com/tags/tag_area.asp) and maybe create the reference based on images where you can click on the images showing pd patches and then the image of the patch itself could be used to navigate between other references...

if hugo would be merged, i don't even know if i could use area and map tags so the ideas might be conflicting... we would need to discuss, (i'm just giving an example) so instead of just going on one way, maybe having more opinions would be interesting, and even in the process we could find more people with different ideas to help

Btw, people at plugdata are updating else documentation, i read something like that... there is an interesting momentum there that maybe we could try to bring to vanilla too, asking if people there would like to help here too

**Comment by Lucarda** [comment]
> about html, i have some ideas about tooltips, for example some section talking about number boxes, if you pass your mouse over the word, it shows an image of a number box (or maybe a gif?) so then we would need to change css, and even think how to deal when the user is on mobile

hmm, no.


> but also i have another idea to use html and tags (https://www.w3schools.com/tags/tag_area.asp) and maybe create the reference based on images where you can click on the images showing pd patches and then the image of the patch itself could be used to navigate between other references...

this is a no go. totally crazy. It will take you 10years to do all the `area` `coords` of just 10 files. then maintain it will be hyper-hard

> if hugo would be merged, i don't even know if i could use and so the ideas might be conflicting... we would need to discuss,

almost no way of doing that in an .md file.

**Comment by abreubacelar** [comment]
see? xD but at least i stated my ideas before knowing the work would be very big

> > but also i have another idea to use html and tags (https://www.w3schools.com/tags/tag_area.asp) and maybe create the reference based on images where you can click on the images showing pd patches and then the image of the patch itself could be used to navigate between other references...
> 
> this is a no go. totally crazy. It will take you 10years to do all the `area` `coords` of just 10 files. then maintain it will be hyper-hard
> 

but for example, pd files have the coords of the objects, and we have https://puredata.info/downloads/patch2svg-plugin that can create an svg from the patch. It isn't possible to automatically merge the two information? I would have an image auto generated from the patch, and the pd file itself states which object is being created, so i can use that to create the link to the right reference, and also generate the size of the object to create the mapping

**Comment by Lucarda** [comment]
> but for example, pd files have the coords of the objects, and we have https://puredata.info/downloads/patch2svg-plugin that can create an svg from the patch. It isn't possible to automatically merge the two information? I would have an image auto generated from the patch, and the pd file itself states which object is being created, so i can use that to create the link to the right reference, and also generate the size of the object to create the mapping

can you make a proof of concept?

you can create any repo you like ;)

**Comment by porres** [comment]
> I remember he talking something like that about your changes, and you just said that you would complete it and also didn't bother about alternatives...

I remember him proposing some kind of automatic extraction and conversion to html and me saying I don't mind doing it manually... it's not like I said "don't do it".

> Btw, people at plugdata are updating else documentation, i read something like that... 

Where? they should NOT do this independently... I remember someone contacting me to offer help and I said ok and that I should take contributions to ELSE's repository, but nothing really happened.

> well, you are right about doing more work... i just don't like the idea of the work not being discussed...

We've been discussing this too much already, in different issues, and things are not moving forward and is getting really tiresome for me... I already asked a few times that I would like to have superpowers to manage this upstream if it is the case, no other discussion is relevant if this is not being decided.

Sorry if this seems I want to claim things need to go my own way because I've been working more on this or threaten I don't do it anymore... I know this doesn't sound good, but lucarda pulled this card first and with this never ending discussion I just felt entitled to do the same thing :) I got more things to do, really...

So, it's come down to this, I don't wanna waste more time or energy on this... just tell me what's gonna happen... A) Can I keep working in my documentation branch with separate manual sections to be merged in it? B) May I be granted with superpowers to work on this on the main branch so it's easier for other people? C) Should I spend my time and energy on other stuff? 

Thanks.

**Comment by Lucarda** [comment]
> We've been discussing this too much already, in different issues, and things are not moving forward and is getting really tiresome for me... I already asked a few times that I would like to have superpowers to manage this upstream if it is the case, no other discussion is relevant if this is not being decided.

> Sorry if this seems I want to claim things need to go my own way because I've been working more on this or threaten I don't do it anymore... I know this doesn't sound good, but lucarda pulled this card first and with this never ending discussion I just felt entitled to do the same thing :) I got more things to do, really...

@porres we shouldn't be discussing your "super-powers" to certain branch (in `main:pure-data`) here in your fork. We should discuss it on `main:pure-data`.

anyway if you had that "super-powers" what would you say about https://github.com/Lucarda/pure-data/tree/hugo ?

**Comment by Lucarda** [comment]
If think I'm not the only one to blame for endless discussions. For me is just normal with *you*.

**Comment by abreubacelar** [comment]
well, sorry for starting this here... but anyway, we now know that it is possible to contribute to porres fork without having to fork it, i will try to do more concrete work and adapt to whatever happens 

> 
> can you make a proof of concept?
> 
> you can create any repo you like ;)

i will try! i just checked and the svgs generated specify already the size of the patch and all the positions of the objects, so for example
```
<svg width='496' height='347' version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink'>
```
496 and 347 are the size of abs~-help.pd
```
<polyline points='264.0 201.0 331.0 201.0 331.0 221.0 264.0 221.0 264.0 201.0' id='.x558ccdf08fe0.t558ccdeb25b0R obj' style='stroke: #000000; stroke-width: 1.0; stroke-linejoin: round; stroke-linecap: square; fill: none;'/>
	<rect x='264.0' y='219.0' width='7.0' height='2.0' id='.x558ccdf08fe0.t558ccdeb25b0o0 outlet' style='stroke: #000000; stroke-width: 1.0; fill: black;'/>
	<rect x='264.0' y='201.0' width='7.0' height='2.0' id='.x558ccdf08fe0.t558ccdeb25b0i0 inlet' style='stroke: #000000; stroke-width: 1.0; fill: black;'/>
	<rect x='324.0' y='201.0' width='7.0' height='2.0' id='.x558ccdf08fe0.t558ccdeb25b0i1 inlet' style='stroke: #000000; stroke-width: 1.0; fill: black;'/>
	<text x='266.0' y='216.0' id='.x558ccdf08fe0.t558ccdeb25b0 obj text' style='font-weight: bold; stroke-width: 0; font-family: DejaVu Sans Mono; font-size: 12px; fill: black;'>metro 100</text>
```
this is a metro 100 on 264.0 200.1 that extends to 331. 221.0, so from this i could make the mappings, and extract the name of the object from text tag
	
i am thinking about using beatifulsoup to parse the svg on python... i will see what i can do, and learn in the process xD

**Comment by Lucarda** [comment]
> i am thinking about using beatifulsoup to parse the svg on python... i will see what i can do, and learn in the process xD

good. but i warn you this will be really crazy in the context of html reference.

**Comment by porres** [comment]
>  We should discuss it on main:pure-data.

it's been discussed there. I had written there again and you might have received a notification, see https://github.com/pure-data/pure-data/pull/1812#issuecomment-1304855293

But that discussion is really convoluted, confusing and tiresomely long. I understood we agreed that having access upstream would be the best option (we were just waiting for iohannes' input). I also thought that in the worst case scenario you were agreeing that it would be ok for me to keep my fork and branch (like I've been doing for years) and not stress IOhannes with this. For reference, see this comment of yours and the one below it https://github.com/pure-data/pure-data/pull/1812#issuecomment-1299646721

IOhannes never gave his word on it, and keep discussing this without his input is a bit pointless. But now out of the blue, you have a problem again with the idea of me keeping a branch in my fork, even when now we have more info that the problems you had have solutions.

> I had asked for pure-data:htmldocs and I got it. But this is now on your side as you only need to merge pure-data:htmldocs to your html branch (but don't do it yet).

So, not only we have the same problem back, I have to answer the same questions I already answered before... and you're giving me quite a weird attitude saying things "hey, it's on your side, why haven't you done this yet"?

I don't like this vibe... I also don't like how things are moving on and back.

I have to say this is all on your "your side" as *you* have decided how things should go. You asked for something, you got it, and you're now calling the shots without having throughly discussed first with others who were also working on the same thing. 

> If think I'm not the only one to blame for endless discussions. For me is just normal with you.

It seems you finally met my limits :) 

I can take the responsibility of my part, but since I'm willing to step out, declare defeat and let go, we may have a winner in this stubbornness competition :) 

I don't need any of this and all this atitude is really stressing me out. 

I feel bad to say I don't want to contribute anymore, but you also played this card first and I really don't feel like contributing in this kind of environment where I feel someone is not thinking of others... where A) something is still being discussed (like our editing access to the main repo), where B) something seemed to have been previously settled and it would be ok for me to keep my fork, but all of a sudden "A)" and "B)" mean nothing and I have to answer the same questions again and hear I'm causing trouble and is "all on my side now"... 

If I'm the problem here, I'm stepping out...

I don't have energy to discuss this anymore and the temperature is really high now. I'm assuming that me working on my fork/branch is not a deal anymore. In this case, tell me if you're willing to insist on granting us editing access of the main repo (I am not asking again). If no answer is given or the outcome is negative, I'll find other things to do.

Cheers

**Comment by Lucarda** [comment]
@porres please relax. I had already told you I am medicated for "paranoid schizophrenia".

You are the boss for .pd files. As for .html files you really don't qualify for that yet.

Is not that i'm the boss of .html files. for many years I have been in charge of that and Miller accepted my work. that's all.

I feel you have a need to be the .html boss. If thats what you need I can't give it to you. If I could I would because I can perfectly talk to you (as we always do).

So please don't get mad because IOhannes or Miller don't give you super-powers. May be if you insist to them you will get it: ask them. 

I don't know if there is a technical way to them to give you control on specific branches.

**Comment by Lucarda** [comment]
If you ask me if I want you to be my boss. I like it.

If you want to have better control of the html manual please try:

    https://github.com/Lucarda/pure-data/tree/hugo
    
if you can do basic work with it, give positive or negative feedback in the PR https://github.com/pure-data/pure-data/pull/1822

please.

**Comment by porres** [comment]
> @porres please relax.

ok, peace!

It's not that I want to be the "boss of html". I've been doing lots of work in my documentation branch and I never thought it would only be a branch for "help files / .pd files" only. I have made changes to the manual over there before and even changes in "errors" from .c files.... 

I would just like to start now a more severe revision and rewriting, by also including references that are pared with my revision of the help files references, add new sections and hopefully manage and merge all this into a single 'documentation' PR (I already said this quite a few times). I don't think it should be nothing over my head to be able to do... it's definitely something I wanna do...

what is your current plan for html files by the way? 

Also, I just don't see the point of involving a third party like IOhannes in the middle of this, having him accept every commit and merge it so someone else can sync to this upstream... we're discussing what is the best workflow if many people were contributing to this, when we don't really have any sort of scenario...

**Comment by porres** [comment]
> If you ask me if I want you to be my boss. I like it.

what?

**Comment by Lucarda** [comment]
Ok. this is whats in my mind:

1. I really don't care at all if i have to push to `pure-data:htmldocs` or `porres:htmldocs`.

1. my problem with your plans on restructuring is that I think you will find it useful to do it in .md.

1. since you don't want to wait anything you are going to do it directly on .html files (harder).

1. then if you do that i'll have to do a lot of hand work updating the .md branch 

do you understand my side?

**Comment by porres** [comment]
> I think you will find it useful to do it in .md.

Me too

> since you don't want to wait anything 

no, I can wait, what's going on, what do I need to wait for?

**Comment by Lucarda** [comment]
> no, I can wait, what's going on, what do I need to wait for?

that the .md branch gets accepted. I'll try (may be you can help) to sell it. it is not a simple stuff. it needs the `hugo` dependency to build and other technical stuff.

**Comment by Lucarda** [comment]
can you wait one week?

**Comment by porres** [comment]
> can you wait one week?

of course.
